SynCommons.pas

```diff
@@ -11303,7 +11303,7 @@ function Hash32(Data: pointer; Len: integer): cardinal; overload;
 // - has less colision than Adler32 for short strings
 // - is faster than CRC32 or Adler32, since use DQWord (128 bytes) aligned read
 // - uses RawByteString for binary content hashing, whatever the codepage is
-function Hash32(const Text: RawByteString): cardinal; overload;
+function Hash32(const Text: RawUTF8): cardinal; overload;

```

Comment:
not acceptable (it breaks existing user code), nor needed RawByteString as input allows to pass a RawUTF8

Reply:
Change is reverted.

Reply:
I still need Hash32(const Text: RawUTF8), so it was written in mORMoti18n.
After SetCurrentLanguage(lngEnglish) method was called, all resourcestring need to be tranlated and method GetText was called. There is an implicit conversion from RawUTF8 to RawByteString and it causes loss of data. Please look at attached call stacks.
1. function Hash32(const Text: RawByteString) was used 

![setcurrentlangugage_en](https://user-images.githubusercontent.com/44496770/47726480-dbf16d80-dc5a-11e8-88f0-f39e90378bc5.png)

2. function Hash32(const Text: RawUTF8) was used
![setcurrentlangugage_en2](https://user-images.githubusercontent.com/44496770/47726578-0d6a3900-dc5b-11e8-809f-adc7073f3013.png)

Reply:
Don't trust the debugger. IMHO this is a debugger issue/bug/feature, which makes on its side a  conversion when displaying the value in the IDE.
There is no actual conversion in the generated assembly when you assign a RawUTF8 to a RawByteString (use Alt-F2 to see it).

Reply:
But Hash32(const Text: RawByteString) called inside TLanguageFile.Translate(var English: string) returns result which is not equal to the hash saved in .messages file and so that TLanguage.FindMessage returns an empty string.
When I use Hash32(const Text: RawUTF8)  both for writing strings to .messages file (inside ExtractAllResources) and inside TLanguageFile.Translate method I'm getting the same results.